infor-design/enterprise
# 5159 - Dropdown filtering & 'change' event behavior improvements

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR fixes two different Dropdown bugs described by #5159:
- In all cases, re-selecting a value from a regular Single-Select dropdown (not Multiselect) would cause the `change` event to fire, even though the value hadn't changed.  The Dropdown has been modified to not re-select and fire change events when the same value is re-selected.
- In cases where a Dropdown is configured with `noSearch: true`, the search input inside the dropdown list is no longer filterable, and is made `readonly`.

**Related github/jira issue (required)**:
Closes #5159 

**Steps necessary to review your pull request (required)**:
Pull, build, run!

_To test re-selection:_
- Open http://localhost:4000/components/dropdown/example-index.html
- Open the dropdown and pick "New York".  A toast should display in response to the change event being fired.
- Open the dropdown and re-pick "New York".  The toast will not re-display (change event is not fired).

_To test filter behavior:_
- Open http://localhost:4000/components/dropdown/example-no-search-lsf.html
- Open the first dropdown and try to use the keyboard to filter within the search input (a-z, 0-9).  Filtering will not occur, but typeahead works as expected (quickly type a valid entry like "R" to highlight the corresponding item in the list while the list is open).
- Also test against http://localhost:4000/components/dropdown/example-index.html, where filtering should be possible.
- NOTE: the "typeahead" behavior in either of these configurations is still allowed.

**Included in this Pull Request**:
